components/charts/chartsjs/BarChart.js

```diff
@@ -21,14 +21,23 @@ const calculateData = (recordKeys, records) => {
   const filterItems = (arr, query) => arr.filter(record => record === query);
   const deathsByDataType = recordKeys.map(key => (!key ? 0 : filterItems(records, key).length));
 
+  // Change background color for current year (i.e. incomplete data)
+  const thisYear = new Date().getFullYear();
+  const colorPalette = recordKeys.map(year => {
+    if (year === thisYear) {
+      return '#919191';
+    }
+    return chartColors[0];
+  });
```

Comment:
I'm a little hesitant to introduce more year-specific code into the `BarChart` component. What do you think about handling this somewhere else and trying to keep `BarChart` more naive about the data being passed in?

Reply:
Hey @nholden , I like the idea about trying to keep `BarChart` naive. My initial thought is to create the color palette array in the `Explore` component based on the chart type and record type and then pass that as a prop to the charting components. What do you think?

We could also just have the color palette entirely defined in the dataset schema. It may become a bit redundant, but it gives more control the the dataset.

I'm curious what your thoughts are about this.

Reply:
I see that we're pulling `chartColors` from [`../../../data/chart_colors`](https://github.com/texas-justice-initiative/website-nextjs/blob/master/data/chart_colors.js), so it feels like the actual colors should come from there. It would be great if we could include keys that we consider "incomplete" in the data we pass along to `BarChart` and then let `BarChart` pull the right color from `chart_colors`. Maybe that's a new prop called `incompleteRecordKeys`? Or maybe there's a more elegant way to handle this.